app/models/update_log.rb

```diff
@@ -1,17 +1,62 @@
 class UpdateLog
+  ####################
+  # CONSTANTS        #
+  ####################
+  MAX_UPDATES_TO_KEEP = 10
 
   def self.setting_record
     Setting.find_or_create_by(key: 'metrics_update')
   end
 
-  def self.log_update(time)
+  def self.log_updates(starttime, endtime)
+    self.add_new_log
+    self.update_data(starttime, endtime)
+    self.log_delay
+  end
+
+  def self.average_delay
     setting = self.setting_record
-    setting.value["last_update"] = time
-    setting.save
+    setting.value['average_delay'] ? setting.value['average_delay']: nil
   end
 
   def self.last_update
-    self.setting_record
-    Setting.find_by(key: 'metrics_update').value['last_update']
+    setting = self.setting_record
+    setting.value['constant_update'] ? setting.value['constant_update'].values.last['end_time'] : nil
   end
+
+  private 
```

Comment:
This won't actually make the class methods private. It's important that the *actually* be private methods, though... the important part is documenting which methods are intended to be called from other classes, and which are not.

To get private class methods, you must either set each one private using `private_class_method`, or use `class << self` to open up the singleton class.